pkg/builtin/multi/lpad.go

```diff
@@ -0,0 +1,140 @@
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend/overload"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/lpad"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+var ArgAndRets_Lpad = []argsAndRet{
+	{[]types.T{types.T_varchar, types.T_int64, types.T_varchar}, types.T_varchar},
+}
+
+func init() {
+	extend.FunctionRegistry["lpad"] = builtin.Lpad
+	for _, item := range ArgAndRets_Lpad {
+		overload.AppendFunctionRets(builtin.Lpad, item.args, item.ret)
+	}
+	extend.MultiReturnTypes[builtin.Lpad] = func(es []extend.Extend) types.T {
+		return getMultiReturnType(builtin.Lpad, es)
+	}
+
+	extend.MultiStrings[builtin.Lpad] = func(es []extend.Extend) string {
+		return fmt.Sprintf("lpad(%s, %s, %s)", es[0], es[1], es[2])
+	}
+	overload.OpTypes[builtin.Lpad] = overload.Multi
+	overload.MultiOps[builtin.Lpad] = []*overload.MultiOp{
+		{
+			Min:        3,
+			Max:        3,
+			Typ:        types.T_varchar,
+			ReturnType: types.T_varchar,
+			Fn: func(vecs []*vector.Vector, proc *process.Process, cs []bool) (*vector.Vector, error) {
+				vs := vecs[0].Col.(*types.Bytes) //Get the first arg
+
+				if !cs[1] && vecs[1].Typ.Oid != types.T_int64 {
+					return nil, errors.New("The second argument of the lpad function must be an int64 constant")
+				}
+				if !cs[2] && vecs[2].Typ.Oid != types.T_varchar {
+					return nil, errors.New("The third argument of the lpad function must be an string constant")
+				}
+				lens := vecs[1].Col.([]int64)
+				padds := vecs[2].Col.(*types.Bytes)
+				for _, num := range lens {
+					if num < 0 {
+						vec, err := process.Get(proc, 24*int64(len(vs.Lengths)), types.Type{Oid: types.T_varchar, Size: 24})
+						if err != nil {
+							return nil, err
+						}
+						nulls.Set(vec.Nsp, vecs[0].Nsp)
+						temp := ""
+						lengths_temp := []uint32{}
+						offsets_temp := []uint32{}
+						for k := 0; k < len(vs.Lengths); k++ {
+							temp += "NULL"
+							lengths_temp = append(lengths_temp, 4)
+							if len(offsets_temp) == 0 {
+								offsets_temp = append(offsets_temp, 0)
+							} else {
+								offsets_temp = append(offsets_temp, offsets_temp[len(offsets_temp)-1]+4)
+							}
+
+						}
+						res := &types.Bytes{
+							Data:    []byte(temp),
+							Lengths: lengths_temp,
+							Offsets: offsets_temp,
+						}
+						vector.SetCol(vec, res)
+						return vec, nil
+					}
+				}
+				//use vecs[0] as return
+				if vecs[0].Ref == 1 || vecs[0].Ref == 0 {
+					vecs[0].Ref = 0
+					temp := lpad.LpadVarchar(vs, lens, padds)
+					vs.Data = make([]byte, len(temp.Data))
+					vs.Lengths = make([]uint32, len(temp.Lengths))
+					vs.Offsets = make([]uint32, len(temp.Offsets))
+					copy(vs.Data, temp.Data)
+					copy(vs.Lengths, temp.Lengths)
+					copy(vs.Offsets, temp.Offsets)
+					return vecs[0], nil
+				}
+
+				vec, err := process.Get(proc, 24*int64(len(vs.Lengths)), types.Type{Oid: types.T_varchar, Size: 24})
+				if err != nil {
```

Comment:
I think the size we need for our new vector here is 24, perhaps I am wrong about this

Reply:
Yeah,We could add WeChat for detail communication more??